homework_02/task_01.py

```diff
@@ -67,40 +67,40 @@ def get_rarest_char(file_path: str, encoding: str = "unicode-escape") -> str:
     """
     Find rarest symbol for document.
     """
-    chars = [
-        symb
-        for token_type, symb in tokenize(file_path, encoding)
-        if token_type == "symbol"
-    ]
-    char_counter = Counter(chars)
-    n_min = len(chars)
-    for symb, n in char_counter.items():
-        if n < n_min:
-            symb_min = symb
-            n_min = n
-    return symb_min
+    char_counter = Counter(
+        [
+            symb
+            for token_type, symb in tokenize(file_path, encoding)
+            if token_type == "symbol"
+        ]
+    )
+    return min(char_counter, key=char_counter.get)
 
 
 def count_punctuation_chars(file_path: str, encoding: str = "unicode-escape") -> int:
     """
     Count every punctuation char.
     """
-    count = 0
-    for token_type, _symb in tokenize(file_path, encoding):
-        if token_type == "punctuation":
-            count += 1
-    return count
+    return len(
+        [
+            token
+            for token_type, token in tokenize(file_path, encoding)
+            if token_type == "punctuation"
+        ]
+    )
 
 
 def count_non_ascii_chars(file_path: str, encoding: str = "unicode-escape") -> int:
     """
     Count every punctuation char.
     """
-    count = 0
-    for token_type, _symb in tokenize(file_path, encoding):
-        if token_type == "non-ascii":
-            count += 1
-    return count
+    return len(
+        [
+            token
+            for token_type, token in tokenize(file_path, encoding)
+            if token_type == "non-ascii"
+        ]
+    )
```

Comment:
sum + generator expression

```suggestion
    return sum(1 for token_type, token in tokenize(file_path, encoding) if token_type == "non-ascii")
```